controller/lib/src/grpc_client/interface.rs

```diff
@@ -0,0 +1,153 @@
+use log::info;
+use proto::scheduler::instance_service_client::InstanceServiceClient;
+use proto::scheduler::node_service_client::NodeServiceClient;
+use proto::scheduler::{
+    Instance, InstanceIdentifier, InstanceStatus, NodeRegisterRequest, NodeStatus,
+    NodeUnregisterRequest, NodeUnregisterResponse,
+};
+use tonic::transport::{Channel, Error};
+use tonic::{Request, Response, Status, Streaming};
+
+pub enum SchedulerClientInterfaceError {
+    ConnectionError(Error),
+    RequestFailed(Status),
+}
+
+pub struct SchedulerClientInterface {
+    node_client: NodeServiceClient<Channel>,
+    instance_client: InstanceServiceClient<Channel>,
+}
+
+impl SchedulerClientInterface {
+    pub async fn new(
+        node_client_address: String,
+        instance_client_address: String,
+    ) -> Result<Self, SchedulerClientInterfaceError> {
+        info!(
+            "Starting gRPC client for scheduler Node Service on {}",
+            node_client_address,
+        );
+
+        let node_client = match NodeServiceClient::connect(node_client_address).await {
+            Ok(client) => client,
+            Err(e) => return Err(SchedulerClientInterfaceError::ConnectionError(e)),
```

Comment:
Please use error mapping here (e.g. `map_err`).

Reply:
Removed in `fixup` commit a5765f3